query-engine/driver-adapters/js/connector-test-kit-executor/src/index.ts

```diff
@@ -0,0 +1,106 @@
+import pgDriver from 'pg'
+import * as pg from '@jkomyno/prisma-adapter-pg'
+import * as qe from './qe'
+import * as engines from './engines/Library'
+import * as readline from 'node:readline'
+import * as jsonRpc from './jsonRpc'
+
+async function main(): Promise<void> {
+    const iface = readline.createInterface({
+        input: process.stdin,
+        output: process.stdout,
+        terminal: false,
+    });
+
+    iface.on('line', async (line) => {
+        try {
+            const request: jsonRpc.Request = JSON.parse(line); // todo: validate
+            console.error(`Got a request: ${line}`)
+            try {
+                const response = await handleRequest(request.method, request.params)
+                respondOk(request.id, response)
+            } catch (err) {
+                respondErr(request.id, {
+                    code: 1,
+                    message: err.toString(),
+                })
+            }
+        } catch (_) {
+            // skip non-JSON line
+        }
+
+    });
+}
+
+const schemas: Record<number, engines.QueryEngineInstance> = {}
+
+async function handleRequest(method: string, params: unknown): Promise<unknown> {
+    switch (method) {
+        case 'initializeSchema': {
+            interface InitializeSchemaParams {
+                schema: string
+                schemaId: string
+                url: string
+            }
+
+            const castParams = params as InitializeSchemaParams;
+            const engine = await initQe(castParams.url, castParams.schema);
+            await engine.connect("")
+            schemas[castParams.schemaId] = engine
+            return null
+        }
+        case 'query': {
+            interface QueryPayload {
+                query: string
+                schemaId: number
+            }
+
+            console.error("Got `query`", params)
+            const castParams = params as QueryPayload;
+            const result = await schemas[castParams.schemaId].query(JSON.stringify(castParams.query), "")
```

Comment:
This is elegant and easy to follow. Each time we are provided a different schema, we initialize a different instance of the query engine, that we disconnect and delete for garbage collection once it's torn down.